packages/monitor-v2/test/utils.ts

```diff
@@ -10,6 +10,13 @@ export async function getContractFactory(contractName: ContractName, signer?: Si
   return new ContractFactory(contractAbi, contractBytecode, signer);
 }
 
-export const { formatBytes32String, parseUnits, toUtf8Bytes, toUtf8String } = utils;
+// Get block number from transaction (or 0 if transaction is not mined).
+export const getBlockNumberFromTx = async (tx: ContractTransaction): Promise<number> => {
+  await tx.wait();
+  const blockNumber = tx.blockNumber ? tx.blockNumber : 0;
```

Comment:
It doesn't really matter here, but in general, prefer !== undefined or !== null when dealing with an optional number to distinguish 0 from non existent/present value.

Reply:
Thanks, fixed in 2e98e4d32533e4c8405d29216fba529bb5049ae4